models/topic.go

```diff
@@ -115,6 +119,24 @@ func (t *Topic) Validate() error {
 	return nil
 }
 
+// Validate checks that a topic update struct complies with the state / release date constraints, if provided.
+func (t *TopicUpdate) ValidateUpdate() error {
```

Comment:
We will want to extend this to check that all fields exist (except keywords) and that they are not empty strings

Once this is done we should handle error message back to API user to inform them that they are missing mandatory data

Reply:
Made other fields mandatory and checked not empty - I think empty subtopics is valid though, so have allowed that. Let me know if you think that's the wrong approach though.